sbi/analysis/plot.py

```diff
@@ -976,3 +976,312 @@ def _get_default_opts():
+                        )
+                elif plot_type == "hist":
+                    plot_ranks_as_hist(
+                        ranki[:, jj],
+                        num_bins,
+                        num_posterior_samples,
+                        label=ranks_labels[ii],
+                        color=f"C{ii}" if colors is None else colors[ii],
+                        xlabel=f"posterior rank {parameter_labels[jj]}",
+                        # Show legend and ylabel only in first subplot.
+                        show_ylabel=False,
+                        show_legend=jj == 0,
+                        alpha=line_alpha,
+                    )
+                    # TODO: fix uniform region plotting.
+                #                     plot_hist_region_expected_under_uniformity(num_sbc_runs,
+                #                                                                num_bins,
+                #                                                                num_posterior_samples,
+                #                                                                alpha=0.1)
+                else:
+                    pass
+                    # TODO: raise ValueError(f"plot_type {plot_type} not defined, use one in {plot_types}")
+
+    # When there is only one set of ranks show all params in a single subplot.
+    else:
+        if fig is None or ax is None:
+            fig, ax = plt.subplots(1, 1, figsize=figsize)
+
+        plt.sca(ax)
+        ranki = ranks[0]
+        for jj in range(num_parameters):
+            plot_ranks_as_cdf(
+                ranki[:, jj],
+                num_bins,
+                num_repeats,
+                label=parameter_labels[jj],
+                color=f"C{jj}" if colors is None else colors[jj],
+                xlabel=f"posterior rank",
+                # Plot ylabel and legend at last.
+                show_ylabel=jj == (num_parameters - 1),
+                show_legend=jj == (num_parameters - 1),
+                alpha=line_alpha,
+            )
+        if show_uniform_region:
+            plot_cdf_region_expected_under_uniformity(
+                num_sbc_runs, num_bins, num_repeats, alpha=uniform_region_alpha
+            )
+
+    return fig, ax
+
+
+def plot_ranks_as_hist(
+    ranks: Tensor,
+    num_bins: int,
+    num_posterior_samples: int,
+    label: str = None,
+    color: str = None,
+    alpha: float = 0.8,
+    xlabel: str = None,
+    show_ylabel: bool = False,
+    show_legend: bool = False,
+    num_ticks: int = 5,
+    xlim_offset_factor: float = 0.1,
+    legend_kwargs: dict = {},
+) -> None:
+    """Plot ranks as empirical CDFs.
+
+    Args:
+        ranks:
+        num_bins:
+        num_repeats:
+        label:
+        color:
+        alpha:
+        xlabel:
+        show_ylabel:
+        show_legend:
+        num_ticks:
+        legend_kwargs:
+
+    """
+    xlim_offset = int(num_posterior_samples * xlim_offset_factor)
+    plt.hist(
+        ranks,
```

Comment:
This current use of ranks (which is likely a `Tensor`) gives me a matplotlib error, I suggest
```suggestion
        ranks.numpy().flatten(),
```